kpet/cmd_tree.py

```diff
@@ -41,7 +45,18 @@ def main(args):
     database = data.Base(args.db)
     if args.action == 'list':
         regex = re.compile(args.regex) if args.regex else None
-        for tree in sorted(database.trees.keys()):
+        arch_regex = re.compile(args.arch) if args.arch else None
```

Comment:
You can simplify this line and the code below if you do this instead:

    arch_regex = re.compile(args.arch or ".*")

The `regex` above could use that change too (in a separate commit preceding this one).